src/Microsoft.PowerShell.IoT/Microsoft.PowerShell.IoT.cs

```
@@ -118,7 +118,17 @@ public GetI2CDevice()
 
         protected override void ProcessRecord()
         {
-            WriteObject(new I2CDevice(Unosquare.RaspberryIO.Pi.I2C.AddDevice(this.Id), this.Id, this.FriendlyName));
+            try
+            {
+                WriteObject(new I2CDevice(Unosquare.RaspberryIO.Pi.I2C.AddDevice(this.Id), this.Id, this.FriendlyName));
+            }
+            catch (System.TypeInitializationException e) // Unosquare.RaspberryIO.Gpio.GpioController.Initialize throws this TypeInitializationException
+            {
+                if (!Unosquare.RaspberryIO.Computer.SystemInfo.Instance.IsRunningAsRoot)
+                {
+                    throw new PlatformNotSupportedException(Resources.ErrNeedRootPrivileges, e);
+                }
```

If they are running as root we should probably `throw` the error again so we don't accidentally swallow it?

---

Good point. Updated here and in other places.